src/analyses/goto_check.cpp

```diff
@@ -907,23 +904,57 @@ void goto_checkt::pointer_overflow_check(
   if(!enable_pointer_overflow_check)
     return;
 
-  if(expr.id()==ID_plus ||
-     expr.id()==ID_minus)
+  if(expr.id()!=ID_plus &&
+     expr.id()!=ID_minus)
+    return;
+
+  if(expr.operands().size()>2)
   {
-    if(expr.operands().size()==2)
+    // The overflow checks are binary!
+    // We break these up.
+
+    for(unsigned i=1; i<expr.operands().size(); i++)
```

Comment:
I needed a pen and paper to fully get what was going on here. I wonder if a comment with an example explaining the calls to pointer_overflow_check would help? 

```C++
// an operand with 4 children, a, b, c, d we:
// first check (a, b)
// then c with a temporary expression with two children: (a, b)
// then d with a temporary expression with three children: (a, b, c)
```